### PR TITLE
Review fixes

### PR DESCRIPTION
**Pull request by psyomn**
Closes #9

### CODE REVIEW

File: service_handler.go [hunk 1]
```diff
@@ -144,22 +146,22 @@ func AddressBookNew(session Session) *AddressBook {
...
+	s.mutex.Lock()
 	rawurl := service.URL.String()
 	if entry, ok := s.entries[rawurl]; ok {
 		if entry.running {
 			entry.Stop()
 		}
 	}
 	s.entries[rawurl] = &*service
-	s.Mutex.Unlock()
+	s.mutex.Unlock()
```
Comment (RAttab) [inline service_handler.go]: I recommend using `defer s.mutex.Unlock()` right after you lock the mutex to make sure you don't forget to unlock it in all code paths.

File: service_handler.go [hunk 2]
```diff
@@ -144,22 +146,22 @@ func AddressBookNew(session Session) *AddressBook {
...
 }
 
 // Get gets a reference to the service with the given rawurl.
 func (s *AddressBook) Get(rawurl string) (*Service, bool) {
-	s.Mutex.Lock()
+	s.mutex.Lock()
 	val, found := s.entries[rawurl]
-	s.Mutex.Unlock()
+	s.mutex.Unlock()
```
Comment (RAttab) [inline service_handler.go]: idem for `defer`

File: service_handler.go [hunk 3]
```diff
@@ -215,21 +217,21 @@ commands:
...
-	s.Mutex.Lock()
+	s.mutex.Lock()
 	if service, ok := s.entries[rawurl]; ok {
 		service.Stop()
 		delete(s.entries, rawurl)
 	} else {
 		log.Print("no such entry to delete: ", rawurl)
 	}
-	s.Mutex.Unlock()
+	s.mutex.Unlock()
```
Comment (RAttab) [inline service_handler.go:228]: idem for using `defer`

File: service_handler.go (hunk 1 above)
Comment (psyomn) [reply service_handler.go]: fixed

File: service_handler.go (hunk 2 above)
Comment (psyomn) [reply service_handler.go]: fixed

File: service_handler.go (hunk 3 above)
Comment (psyomn) [reply service_handler.go:228]: Fixed.

**Comment by psyomn** [comment]
Thank you!

**Pull request merged**